src/LocationCamera.cpp

```diff
@@ -21,93 +21,55 @@
 
 // Falltergeist includes
 #include "LocationCamera.h"
+#include "Logger.h"
 
 // Third party includes
 
 namespace Falltergeist
 {
 
-LocationCamera::LocationCamera(int width, int height, int xPosition, int yPosition)
-{
-    setWidth(width);
-    setHeight(height);
-    setXPosition(xPosition);
-    setYPosition(yPosition);
-}
-
-LocationCamera::~LocationCamera()
-{
-}
 
-int LocationCamera::x()
+LocationCamera::LocationCamera(const Size& size, const Point& center)
 {
-    return xPosition() - width()*0.5;
+    setSize(size);
+    setCenter(center);
 }
 
-
-int LocationCamera::y()
-{
-    return yPosition() - height()*0.5;
-}
-
-int LocationCamera::xPosition()
+LocationCamera::~LocationCamera()
 {
-    return _xPosition;
 }
 
-void LocationCamera::setXPosition(int value)
+Point LocationCamera::topLeft() const
 {
-    if (value < width()*0.5)
-    {
-        _xPosition = width()*0.5;
-    }
-    else
-    {
-        _xPosition = value;
-    }
+    return _center - (_size / 2.0);
 }
 
-int LocationCamera::yPosition()
+Point LocationCamera::center() const
 {
-    return _yPosition;
+    return _center;
 }
 
-void LocationCamera::setYPosition(int value)
+void LocationCamera::setCenter(const Point& pos)
 {
-    if (value < height()*0.5)
+    _center = pos;
+    if (_center.x() < _size.width() / 2)
     {
-        _yPosition = height()*0.5;
+        _center.setX(_size.width() / 2);
     }
-    else
+    if (_center.y() < _size.height() / 2)
     {
-        _yPosition = value;
+        _center.setY(_size.height() / 2);
     }
 }
 
-void LocationCamera::setPosition(int x, int y)
-{
-    setXPosition(x);
-    setYPosition(y);
-}
-
-int LocationCamera::width()
-{
-    return _width;
-}
-
-void LocationCamera::setWidth(int value)
-{
-    _width = value;
-}
-
-int LocationCamera::height()
+Size LocationCamera::size() const
```

Comment:
For performance? What is your rationale and do you think do the same for other classes?